controller/instance_manager_controller.go

```diff
@@ -2231,7 +2256,7 @@ func (m *InstanceManagerMonitor) isEngineOrphaned(instanceName, node string) (bo
 	switch {
 	case err == nil:
 		// Engine CR still exists - check the ownership
-		return m.isInstanceOrphanedOnNode(&existEngine.Spec.InstanceSpec, &existEngine.Status.InstanceStatus, node), nil
+		return m.isInstanceOrphanedOnNode(&existEngine.ObjectMeta, &existEngine.Spec.InstanceSpec, &existEngine.Status.InstanceStatus, node), nil
```

Comment:
Wht not directly pass engine to `isInstanceOrphanedOnNode`?

Reply:
`isInstanceOrphanedOnNode` is shared between `v1beta2.Engine` and `v1beta2.Replica`. There's no common interface between these two to retrieve these three embedded structure.